analytics.js

```diff
@@ -833,11 +854,16 @@ class Analytics {
       checkReservedKeywords(rudderElement.message, type);
 
       // if not specified at event level, All: true is default
-      const clientSuppliedIntegrations = rudderElement.message.integrations || { 'All' : true };
+      const clientSuppliedIntegrations = rudderElement.message.integrations || {
+        All: true,
+      };
 
       // structure user supplied integrations object to rudder format
       tranformToRudderNames(clientSuppliedIntegrations);
-      rudderElement.message.integrations = clientSuppliedIntegrations;
+      rudderElement.message.integrations = {
+        ...this.integrationsData,
+        ...clientSuppliedIntegrations,
+      };
```

Comment:
Simple object merge will not work for all the cases. You should use the lodash merge utility.

ex:
```
integrationsData = {
  "Google Analytics 4": {
    "sessionId": "123"
  }
}
------------------------------

rudderanalytics.track("Some event", {}, {
  integrations: {
    All: true
    "Google Analytics 4": {
      "some other data": "abc"
    }
  }
});
------------------------------
This will result in:
{
  type: "track",
  event: "Some event",
  ...
  integrations: {
    All: true
    "Google Analytics 4": {
      "some other data": "abc"
    }
  }
}
but what we want is:
{
  type: "track",
  event: "Some event",
  ...
  integrations: {
    All: true
    "Google Analytics 4": {
      "sessionId": "123",
      "some other data": "abc"
    }
  }
}

```
CC @MoumitaM

Reply:
updated